File: LAB DSA/min_max.java
public class min_max {
    static int minmax(int arr[],int start, int end, boolean isMax){
        if(start==end){
            return arr[start];
        }
        //divide
        int mid=(start+end)/2;
        //conquerr
        int left=minmax(arr, start, end, !isMax);
        int right=minmax(arr, mid+1, end, !isMax);
        //combine
        if(isMax){
            if(left>right)
                return left;
            
            else
                return right;
        }
            else
                if(left<right)
                    return left;
                else
                    return right;
            }

       
    public static void main(String[] args) {
        int arr[]={3,2,5,9};
        int result=minmax(arr, 0, arr.length-1,true);
        System.out.println("answer:- "+result);
    }}

File: LAB DSA/min_max.c
#include <stdio.h>
#include <stdbool.h>

int minmax(int arr[], int start, int end, bool isMax) {
    if (start == end) {
        return arr[start];
    }

    // divide
    int mid = (start + end) / 2;

    // conquer
    int left = minmax(arr, start, mid, !isMax);
    int right = minmax(arr, mid + 1, end, !isMax);

    // combine
    if (isMax) {
        if (left > right)
            return left;
        else
            return right;
    } else {
        if (left < right)
            return left;
        else
            return right;
    }
}

int main() {
    int arr[] = {1, 2, 5, 9};
    int n = sizeof(arr) / sizeof(arr[0]);

    int result = minmax(arr, 0, n - 1, true);
    int result2 = minmax(arr, 0, n - 1, false);

    printf("answer:- %d\n", result);
        printf("answer:- %d\n", result2);

    return 0;
}